core/src/main/scala/org/bitcoins/core/protocol/tlv/TLV.scala

```diff
@@ -1172,6 +1240,55 @@ object DLCOfferTLV extends TLVFactory[DLCOfferTLV] {
   }
 }
 
+sealed trait NegotiationFieldsTLV extends TLV
+
+object NegotiationFieldsTLV extends TLVParentFactory[NegotiationFieldsTLV] {
+
+  override val allFactories: Vector[TLVFactory[NegotiationFieldsTLV]] =
+    Vector(NoNegotiationFieldsTLVFactory, NegotiationFieldsV1TLV)
+
+  override val typeName: String = "NegotiationFieldsTLV"
+}
+
+case object NoNegotiationFieldsTLV extends NegotiationFieldsTLV {
+  override val tpe: BigSizeUInt = NoNegotiationFieldsTLVFactory.tpe
+
+  override val value: ByteVector = ByteVector.empty
+}
+
+object NoNegotiationFieldsTLVFactory
+    extends TLVFactory[NoNegotiationFieldsTLV.type] {
+  override val tpe: BigSizeUInt = BigSizeUInt(55334)
+
+  override def fromTLVValue(value: ByteVector): NoNegotiationFieldsTLV.type = {
+    require(value.isEmpty, "NoNegotiationsFieldsTLV must be empty")
+
+    NoNegotiationFieldsTLV
+  }
+}
```

Comment:
nit: imo `EmptyNegotiationFieldsTLV` would be a better name, but I am indifferent